test/uk/gov/hmrc/transitmovementsauditing/services/AuditServiceSpec.scala

```diff
@@ -72,6 +72,9 @@ class AuditServiceSpec
       |  "messageSender":
       |}""".stripMargin
 
+  private val someValidDetailsJson =
+    """{"metadata":{"path":"some-path","movementId":"movementId","messageId":"messageId","enrolmentEORI":"enrolmentEORI","movementType":"departure","messageType":"IE015"},"payload":{"messageSender":"sender1"}}""".stripMargin
+
```

Comment:
This applies to all the hand-written Json but... why not just create a `Details` object, because all you ever do is parse a `Details` object out of this, then do a `toString` somewhere to do an equality check. We should just keep things as `Details` in this case, using `Json.toJson` where necessary.

A test like this is kind of okay, but brittle, because the `toString` representation could change.

With that in mind, as well as this minimal and full test (with tweaks), we really should use property based testing and just throw random `Details` objects in using `forAll`, because the reality is we don't really care about what the contents of the details object is, just that what we get in is what we get out.

Reply:
yes, used the Details obj now